File: code-samples/voice/user-answering-rules.java
import com.ringcentral.*;
import com.ringcentral.definitions.*;

public class UserCustomAnsweringRule {
	public static void main(String[] args) {
		try {
			create_user_custom_answering_rule();
		} catch (RestException | IOException e) {
			e.printStackTrace();
		}
	}

	public static void create_user_custom_answering_rule() throws RestException, IOException {
		RestClient rc = new RestClient("client_id", "client_secret", "server_url");
		rc.authorize("username", "extension_number", "password");

		var parameters = new CreateUserAnsweringRuleRequest();
		parameters.enabled = true;
		parameters.type = "Custom";
		parameters.name = "My weekly meetings";
		var schedule = new ScheduleInfo();
		var weeklyRanges = new WeeklyScheduleInfo();
		TimeInterval meetingTime = new TimeInterval();
		meetingTime.from = "09:00";
		meetingTime.to = "10:00";
		weeklyRanges.monday = new TimeInterval[] { meetingTime };

		meetingTime = new TimeInterval();
		meetingTime.from = "10:00";
		meetingTime.to = "15:00";
		weeklyRanges.friday = new TimeInterval[] { meetingTime };

		schedule.weeklyRanges = weeklyRanges;
		parameters.schedule = schedule;
		parameters.callHandlingAction = "TakeMessagesOnly";

		var response =  rc.restapi().account().extension().answeringrule().post(parameters);
		System.out.println(JSON.toJSONString(response));
	}
}

File: code-samples/voice/user-answering-rules.php
<?php
require('vendor/autoload.php');
$dotenv = Dotenv\Dotenv::createImmutable(__DIR__ . '/../');
$dotenv->load();

$CLIENTID     = $_ENV['RC_CLIENT_ID'];
$CLIENTSECRET = $_ENV['RC_CLIENT_SECRET'];
$SERVER       = $_ENV['RC_SERVER_URL'];
$USERNAME     = $_ENV['RC_USERNAME'];
$PASSWORD     = $_ENV['RC_PASSWORD'];
$EXTENSION    = $_ENV['RC_EXTENSION'];

$rcsdk = new RingCentral\SDK\SDK($CLIENTID, $CLIENTSECRET, $SERVER);
$platform = $rcsdk->platform();
$platform->login($USERNAME, $EXTENSION, $PASSWORD);

$params = array (
    'enabled' => true,
    'type' => "Custom",
    'name' => "My weekly meetings",
    'schedule' => array (
      'weeklyRanges' => array (
	'monday' => array ( array ('from' => "09:00", 'to' => "10:00")),
	'friday' => array ( array ('from' => "10:00", 'to' => "15:00"))
      )
    ),
    'callHandlingAction' => "TakeMessagesOnly"
);
$resp = $platform->post('/account/~/extension/~/answering-rule', $params);

print_r ($resp->text());
?>